assembly/src/ast/program.rs

```diff
@@ -327,3 +327,37 @@ impl fmt::Display for ProgramAst {
         writeln!(f, "end")
     }
 }
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for ProgramAst {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        self.body.write_into(target);
+        target.write_u64(self.local_procs.len() as u64);
```

Comment:
We should be able to use `u16` here as the number of local procs is limited to `u16::MAX`.

Reply:
changed